src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -58,20 +112,103 @@ private object LoadOptionFromSerializerOrGetDefault(OptionKey optionKey)
 
         public IEnumerable<IOption> GetRegisteredOptions()
         {
-            return _options.Value;
+            return _lazyAllOptions.Value;
         }
 
+        public ImmutableHashSet<IOption> GetRegisteredSerializableOptions(ImmutableHashSet<string> languages)
+        {
+            if (languages.IsEmpty)
+            {
+                return ImmutableHashSet<IOption>.Empty;
+            }
+
+            var builder = ImmutableHashSet.CreateBuilder<IOption>();
+
+            // "string.Empty" for options from language agnostic option providers.
+            builder.AddRange(GetOrComputeSerializableOptionsForLanguage(string.Empty));
+
+            foreach (var language in languages)
+            {
+                builder.AddRange(GetOrComputeSerializableOptionsForLanguage(language));
+            }
+
+            return builder.ToImmutable();
+        }
+
+        /// <summary>
+        /// Gets force computed serializable options with prefetched values for all the registered options applicable to the given <paramref name="languages"/> by quering the option persisters.
+        /// </summary>
+        public SerializableOptionSet GetForceComputedOptions(ImmutableHashSet<string> languages, IOptionService optionService)
+        {
+            var serializableOptionKeys = GetRegisteredSerializableOptions(languages);
+            var serializableOptionValues = GetSerializableOptionValues(serializableOptionKeys, languages);
+            return new SerializableOptionSet(languages, optionService, serializableOptionKeys, serializableOptionValues);
+        }
+
+        private ImmutableDictionary<OptionKey, object?> GetSerializableOptionValues(ImmutableHashSet<IOption> optionKeys, ImmutableHashSet<string> languages)
+        {
+            if (optionKeys.IsEmpty)
+            {
+                return ImmutableDictionary<OptionKey, object?>.Empty;
+            }
+
+            ForceComputeOptionValues(optionKeys, languages);
+
+            lock (_gate)
+            {
+                return ImmutableDictionary.CreateRange(_currentValues
+                    .Where(kvp => optionKeys.Contains(kvp.Key.Option) &&
+                                   (!kvp.Key.Option.IsPerLanguage ||
+                                    languages.Contains(kvp.Key.Language!))));
+            }
+
+            // Local functions
+            void ForceComputeOptionValues(ImmutableHashSet<IOption> options, ImmutableHashSet<string> languages)
```

Comment:
>ForceComputeOptionValues [](start = 17, length = 24)

Inline?

Reply:
It seemed much cleaner this way as this method has an early bail out.

Reply:
Inlined.